### ISSUE TITLE
RFE - Procreation / Marriage enhancements

### ISSUE THREAD

**Issue by mjbroekman** [MegaMek/mekhq#2394]
This is, admittedly, far-future but I'm submitting it in case someone wants to pick it up before Windchild does.

### Description
Currently, Use Procreation Without a Relationship allows children to be born with no father set (aka, father is some rando of no note and not in the Roster). This is currently working as intended.

The enhancement would work as follows:
1. Add options for:
- 'Always Determine Father' to force the Father to be a living member of the Roster (including Prisoners)
- 'Allow Polygamy' to allow multiple simultaneous living spouses
- 'Allow Infidelity' to only allow married personnel to procreate with unmarried personnel (or non-fathers)
- 'Allow Divorce' to allow spouses to separate (higher chances if spouses have children with non-spouses)

2. Adjust chance of 'non-Roster' father's based on contract type
- Example: if you are in hostile territory, there are severely reduced chances for a 'non-Roster' Father.
- Example: if you are on a Garrison Duty or Cadre Duty contract, there are elevated chances for a 'non-Roster' Father.

3. With 'Allow polygamy', allow a chance of the Father to be a male that already has a different Spouse
4. Increase chances of Mother and Father getting married after conception and birth if they are not married.
5. When determining the father at conception (or birth), if a non-roster father is "generated", give a small chance that the next month will add a male Dependent to the roster that is (becomes) the spouse and father. Re-check every month until birth. After birth, no more checks occur and the child does not know their father.

**Comment by binaryspica** [comment]
An another idea is, what about to allow procreates twins? I remember that it is possible to randomly had a twins, but it is not possible manually.

**Comment by Windchild292** [comment]
The manual pregnancy command makes the same random roll, 1/50 chance of having more than one child, with it continuing to roll up to a maximum of decuplets. Adding anything other than the current setup would be very cluttering.

**Comment by binaryspica** [comment]
Yeah it would be cluttering. If it should be implemented, then it would be better to modify the rate of having more than one child on the personnel option instead.

**Comment by Windchild292** [comment]
That I can easily do once #2417 has been merged.

**Comment by binaryspica** [comment]
After all you don't need to make a twins too often, so I think that it would be enough. And it would be interesting to change the odds of having a twins in the random procreation as well.

**Comment by FastJack-MD** [comment]
So, Polyamory is being worked on for the game, but my request for [Gender Identities](https://github.com/MegaMek/mekhq/issues/876) was closed because "It doesn't impact game mechanics".

Got it.

**Comment by Tzahr** [comment]
> So, Polyamory is being worked on for the game, but my request for [Gender Identities](https://github.com/MegaMek/mekhq/issues/876) was closed because "It doesn't impact game mechanics".
> 
> Got it.

Your previoues request is more than five years old, and handled by a different developer than the one who implemented these current features.

The 'Other' gender identity was also implemented recently and should be available in future builds - if that does not suffice to fulfill your previous request, you can always try filing it as a new RFE and request additional implementation.

**Comment by IllianiCBT** [comment]
This is a feature request, not a development post. This is another user saying "hey I want this", not a dev saying "this is being implemented". Polyamorous relationships are not being implemented.

**Comment by IllianiCBT** [comment]
Also, non-binary genders have already been implemented into 50.01, which I vaguely remember being the core focus of your request.
